### PR TITLE
introduce a graphql job

### PR DESCRIPTION
**Pull request by nlandolfi**
Introduce a graphql deployment. 

The meat is in jobs/graphql/main.go where I define a lot of the schema.

 @viggyfresh we should talk more about what you envision our query and mutation objects looking like.

### CODE REVIEW

File: jobs/bump.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+if ! ($SED --version 2>&1 | grep -q GNU); then
+  echo "!!! GNU sed is required.  If on OS X, use 'brew install gnu-sed'."
+  exit 1
+fi
+
+cd $(dirname $0)
+
+makefile_version_re="^\(VERSION.*=\s*\)"
+version=$($SED -n "s/$makefile_version_re//Ip" ${1}/Makefile)
+new_version=$(awk -F. '{print $1 "." $2+1}' <<< $version)
```
Comment (viggyfresh) [inline jobs/bump.sh]: LOL

File: jobs/graphql/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: apps/v1beta1 # for versions before 1.6.0 use extensions/v1beta1
```
Comment (viggyfresh) [inline jobs/graphql/deployment.yaml]: why v1beta1?

File: jobs/ingress.yaml [hunk 3]
```diff
@@ -19,8 +19,9 @@ spec:
...
 metadata:
-  name: cerberus
+  name: graphql
 spec:
   backend:
-    serviceName: cerberus
-    servicePort: 3333
+    serviceName: graphql
+    servicePort: 8005
+---
```
Comment (viggyfresh) [inline jobs/ingress.yaml]: intentional?

File: jobs/ingress.yaml [hunk 4]
```diff
@@ -19,8 +19,9 @@ spec:
 apiVersion: extensions/v1beta1
 kind: Ingress
 metadata:
-  name: cerberus
+  name: graphql
```
Comment (viggyfresh) [inline jobs/ingress.yaml]: also intentional?

**Comment by viggyfresh** [comment]
feel free to squash and merge when you're ready!

File: jobs/graphql/main.go [hunk 5]
```diff
@@ -0,0 +1,1745 @@
...
+})
+
+var modelsCredential = gq.NewObject(gq.ObjectConfig{
+	Name: "models_Credential",
+	Fields: gq.Fields{
+		"id":                   &gq.Field{Type: gq.NewNonNull(gq.String)},
+		"type":                 &gq.Field{Type: modelsCredentialTypeEnum},
+		"public":               &gq.Field{Type: gq.String},
+		"private":              &gq.Field{Type: gq.String},
+		"owner_id":             &gq.Field{Type: gq.String},
```
Comment (viggyfresh) [inline jobs/graphql/main.go]: how do we handle camelCasing?

File: jobs/graphql/deployment.yaml (hunk 2 above)
Comment (nlandolfi) [reply jobs/graphql/deployment.yaml]: I believe kubernetes "deployments" are technically in beta.

File: jobs/ingress.yaml (hunk 4 above)
Comment (nlandolfi) [reply jobs/ingress.yaml]: Yes intentional because I never successfully exposed cerberus.

File: jobs/ingress.yaml (hunk 3 above)
Comment (nlandolfi) [reply jobs/ingress.yaml]: Yes intentional because I never successfully exposed cerberus.

File: jobs/graphql/main.go (hunk 5 above)
Comment (nlandolfi) [reply jobs/graphql/main.go]: do you think we should change it to be camel-cased? What do you think re: the type names?

e.g., `base_Timestamp`

**Comment by nlandolfi** [comment]
ok we can't have underscores for now because in the generated protobuf structs the json key is set to be the underscore form, and this is what the graphql lib looks at for automatically unwrapping the fields.

https://github.com/elos/x/tree/graphqlCamel

here is the diff branch for  when we can do it.

again recall that we might just move to a different graphql pkg altogether namely: https://github.com/neelance/graphql-go

**Comment by nlandolfi** [comment]
@viggyfresh PTAL and squash and merge whenever. This is deployed at head.

also we have a readme now

File: jobs/graphql/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+It exposes `/` which serves GraphiQL and `/graphql` which can accept the `query` param or alternatively a `{query: ..., variables}` payload (as graphql sends).
+
+To use GraphiQL, go to `http://stg-hyperion.elos.pq` and login, then take your access token and in your variables add:
+```
+{
+    "access_token": "whatever it is here"
+}
+```
+Notice that it is `access_token` snake case. Additioanlly any of hte other methods of providing authentication credentials for elos are supported here. They are all superseded if you supply the `access_token` query variable. For example, if you supplied basic auth and the access_token variable, we use the access_token variable and ignore the basic auth.
```
Comment (viggyfresh) [inline jobs/graphql/README.md]: typos - additionally, the

File: jobs/graphql/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+```
+query {
+    records(query: {kind: TASK}){
+        task {
+            id
+        }
+    }
+}
+```
+alias for an underscore field when we want a camelcase on the frontent
```
Comment (viggyfresh) [inline jobs/graphql/README.md]: underscore -> snake_case, camelcase -> camelCase, frontent -> frontend

**Comment by nlandolfi** [comment]
@viggyfresh fixed PTAL

**Pull request merged**